src/BaseType.php

```diff
@@ -29,6 +29,15 @@ public function setProperty(string $property, $value)
         return $this;
     }
 
+    public function addProperties(array $properties)
```

Comment:
Can this be renamed to `setProperties` for consistency with `setProperty`?

Reply:
I'd say `setProperties` is semantically not what happens here. I would think, that a `setProperties` call resets everything done before vie individual `setProperty` calls.

Reply:
Fair point! Leave it as is then :)